src/sagemaker/lineage/query.py

```diff
@@ -226,29 +226,102 @@ def __init__(
-                    'arn': 'string', 'lineage_entity': 'string', 
-                    'lineage_source': 'string', 
+                    'arn': 'string', 'lineage_entity': 'string',
+                    'lineage_source': 'string',
                     '_session': <sagemaker.session.Session object>
                 },
                 ...
             ]
         }
-        
+
         """
         result_dict = vars(self)
-        return (str({k: [vars(val) for val in v] for k, v in result_dict.items()}))
+        return str({k: [vars(val) for val in v] for k, v in result_dict.items()})
+
+    def _import_visual_modules(self):
+        """Import modules needed for visualization."""
+        import dash_cytoscape as cyto
+
+        from jupyter_dash import JupyterDash
+
+        from dash import html
+
+        return cyto, JupyterDash, html
+
+    def _get_verts(self):
+        """Convert vertices to tuple format for visualizer"""
+        verts = []
+        for vert in self.vertices:
+            verts.append((vert.arn, vert.lineage_source))
+        return verts
+
+    def _get_edges(self):
+        """Convert edges to tuple format for visualizer"""
+        edges = []
+        for edge in self.edges:
+            edges.append((edge.source_arn, edge.destination_arn))
+        return edges
+
+    def visualize(self):
+        """Visualize lineage query result."""
+
+        cyto, JupyterDash, html = self._import_visual_modules()
+
+        cyto.load_extra_layouts()  # load "klay" layout (hierarchical layout) from extra layouts
+        app = JupyterDash(__name__)
+
+        verts = self._get_verts()
+        edges = self._get_edges()
+
+        nodes = [
+            {
+                "data": {"id": id, "label": label},
+            }
+            for id, label in verts
+        ]
+
+        edges = [{"data": {"source": source, "target": target}} for source, target in (edges)]
+
+        elements = nodes + edges
+
+        app.layout = html.Div(
+            [
+                cyto.Cytoscape(
+                    id="cytoscape-layout-1",
+                    elements=elements,
+                    style={"width": "100%", "height": "350px"},
+                    layout={"name": "klay"},
+                    stylesheet=[
+                        {"selector": "node", "style": {"label": "data(label)", "font-size": "7px"}},
+                        {
+                            "selector": "edge",
+                            "style": {
+                                "width": "1px",
+                                "curve-style": "straight",
+                                "target-arrow-color": "black",
+                                "target-arrow-shape": "triangle",
+                                "line-color": "black",
+                            },
+                        },
+                    ],
+                    responsive=True,
+                )
+            ]
+        )
+
+        return app.run_server(mode="inline")
```

Comment:
Out of curiosity, what are the other modes you could use here?

Reply:
There are three modes for run_server -> "inline", "external", and "jupyterlab". The ports can also be specified (default = 8050) in the run_server(field). More detailed JupyterDash info are listed here: https://via.hypothes.is/https://medium.com/plotly/introducing-jupyterdash-811f1f57c02e